roomserver/storage/shared/storage.go

```diff
@@ -124,7 +124,15 @@ func (d *Database) StateEntriesForTuples(
 }
 
 func (d *Database) RoomInfo(ctx context.Context, roomID string) (*types.RoomInfo, error) {
-	return d.RoomsTable.SelectRoomInfo(ctx, roomID)
+	if roomInfo, ok := d.Cache.GetRoomInfo(roomID); ok {
```

Comment:
Is this safe? The room nid and room version may be immutable but the state snapshot nid is not.

Reply:
If the roominfo is present in the cache, then the latest events updater will keep it up-to-date: https://github.com/matrix-org/dendrite/pull/1646/files#diff-e863b7b084310a691d13d9747108c96f6fac397eb7c09411c770aa7c83b4b0ffR108-R114

It has to be a mutable cache for that reason (that and the room can suddenly become not-a-stub when we start updating the latest events).

Reply:
Also this is safe because this cache is strictly used only in the roomserver and not cross-component.

Reply:
Adding comments to this effect.